File: isAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        // we are going to create a unorderd map of the characters in s and then compare it to t
        unordered_map<char, int> sMap;
        unordered_map<char, int> tMap;
        // but first we are going to compare the lengths of the strings
        if (s.length() != t.length()) {
            return false;
        }
        // now we are going to create the map for s
        for (int i = 0; i < s.length(); i++) {
            if (sMap.find(s[i]) == sMap.end()) {
                sMap[s[i]] = 1;
            } else {
                sMap[s[i]]++;
            }
        }
        // now we are going to create the map for t
        for (int i = 0; i < t.length(); i++) {
            if (tMap.find(t[i]) == tMap.end()) {
                tMap[t[i]] = 1;
            } else {
                tMap[t[i]]++;
            }
        }
        // now we are going to compare the maps
        for (auto it = sMap.begin(); it != sMap.end(); it++) {
            if (tMap.find(it->first) == tMap.end()) {
                return false;
            } else if (tMap[it->first] != it->second) {
                return false;
            }
        }
        return true;
    }
};

File: isAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # if the length of the two strings are not equal, return False
        if len(s) != len(t):
            return False
        # than we going to create a dictionary to store the characters in the string
        dict = {}
        # we are going to loop through the string s and store the characters in the dictionary
        for i in s:
            if i in dict:
                dict[i] += 1
            else:
                dict[i] = 1
        # we are going to loop through the string t and check if the characters are in the dictionary
        for i in t:
            if i in dict:
                dict[i] -= 1
            else:
                return False
        # we are going to loop through the dictionary and check if the value is 0
        for i in dict:
            if dict[i] != 0:
                return False
        return True
    
